server/routes/user.js

```diff
@@ -214,6 +214,66 @@ router.get("/getPatients", async (req, res, next) => {
   });
 });
 
+// get alarming patients for specific doctor
+router.get("/getAlarmingPatients", async (req, res, next) => {
+  const doctorLastName = req.query.lastName.toLowerCase();
+
+  const dataLogs = Parse.Object.extend("DataLog");
+  var query = new Parse.Query(dataLogs);
+
+  //find all datalogs with alarming hsl values
+  let alarmingPatientsInfo = [];
+
+  query.find().then((dataLogs) => {
+    dataLogs.map((dataLog) => {
+      let hsl = dataLog.get("drainHSL");
+      let hslArray = hsl.split(", ");
+
+      let h = Number(hslArray[0].replace("(", ""));
+      let s = Number(hslArray[1]);
+      let l = Number(hslArray[2].replace(")", ""));
```

Comment:
I did it this way because right now I have two ways for users to pick colors. Either 
1. they will choose a color from the color picker which will produce a hex code and hsl value (formatted "(h, s, l)") that are both saved in state variables 
or 
2. users will upload a picture of their drain output so the colors can be extracted which will produce a hex code

I wanted the formatting to be the same between hsl values, so I created a string. In retrospect, instead of saving the hsl value that the color picker produced, I could have inputted the hex code into the function and returned a hsl tuple.